libopenage/CMakeLists.txt

```diff
@@ -108,6 +109,91 @@ if(WANT_BACKTRACE)
 	find_package(GCCBacktrace)
 endif()
 
+#######################################
+# nyan integration
+
+# first, try to locate nyan directly
+# this discovers the system package or the user-registry package
+find_package(nyan CONFIG)
+
+# if this didn't work, we can download nyan like a git submodule.
+# this is the treeish to be checked out.
+set(nyan_clone_version origin/master)
+
+option(
+	DOWNLOAD_NYAN
+	"whether to clone the nyan project in case it is not found"
+	OFF
+)
+
+option(
+	FORCE_DOWNLOAD_NYAN
+	"Force the download and usage of the nyan project"
+	OFF
+)
+
+option(
+	DISABLE_SUBPROJECT_UPDATES
+	"Disable the automatic update of subprojects over the internet"
+	OFF
+)
+
+# check if nyan was used as subproject before (and still exists)
+if(DEFINED NYAN_SUBPROJ_LOCATION AND IS_DIRECTORY "${NYAN_SUBPROJ_LOCATION}")
+	set(USED_NYAN_SUBPROJ TRUE)
+else()
+	set(USED_NYAN_SUBPROJ FALSE)
+	set(NYAN_SUBPROJ_LOCATION "" CACHE INTERNAL
```

Comment:
OIh, that's right! So even if the folder is not there any more, it would be downloaded again. and this hopefully is what the user expects.

I'll remove this then, because if I reset `DOWNLOAD_NYAN`, it will never be downloaded in the first place.